frames/frames.go

```diff
@@ -0,0 +1,119 @@
+// Package frames contains an API for interacting with i-frames and representing them into some data structure.
+package frames
+
+import (
+	"image"
+	"io"
+
+	// Due to the way the image package works, we need to register the png import as well, even if it isn't used directly
+	_ "image/png"
+)
+
+// CompressionType represents the type of Compression used on the video file.
+type CompressionType int
+
+const (
+	// MPEG2 is used for files saved with .mp2 and have i-frames every 0.5 seconds.
+	MPEG2 CompressionType = iota
+	// MPEG4 is used for files saved with .mp4 and have i-frames every 0.20002 seconds.
+	MPEG4
+)
+
+// Processor is an interface for processing i-frames into exported Frame structures. The interface requires a single
+// method, Process, which accepts a filename and returns a Frame type to represent it.
+type Processor interface {
+	Process(io.Reader) (uint32, uint32, uint32, uint32, error)
+}
+
+// Frame represents an i-frame by maintaining a single value for each color channel and the number of frame.
```

Comment:
Are you referring to the number of frame or the frame number within the video? The explanation is different between lines 28 and 30.

Reply:
I mean to say the number we extracted. That is, if we grab every 10th frame, the first frame would be 0, the 10th frame would 1 and so on. I'll fix the comments